server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java

```diff
@@ -97,13 +109,21 @@ public Message fromMetaDataWithContent(MetaDataWithContent message) throws Mailb
                 .replyTo(fromAddressList(mimeMessage.getReplyTo()))
                 .size(message.getSize())
                 .date(message.getInternalDateAsZonedDateTime())
-                .textBody(messageContent.getTextBody().orElse(null))
-                .htmlBody(messageContent.getHtmlBody().orElse(null))
-                .preview(getPreview(messageContent))
+                .textBody(textBody)
```

Comment:
You might need Message.builder() accept htmlBody(Optional<String>) and textBody(Optional<String>). (Message supports Optional so it's just adding constructor methods.

We should let INBOX render htmlBody when not here. We should let them render htmlBody and textBody when none of them is specified.